pkg/webhook/pod_webhook.go

```diff
@@ -17,62 +17,213 @@ limitations under the License.
 package webhook
 
 import (
-	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	"fmt"
+	"strings"
 
 	kueue "github.com/openshift/kueue-operator/pkg/apis/kueueoperator/v1alpha1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// arrays for the list of webhooks we remove if pod integration is not needed
-// Kueue builds statefulset and deployment off on pod integration
 var (
-	podBasedValidatingWebhooks = []string{"vdeployment.kb.io", "vpod.kb.io", "vstatefulset.kb.io"}
-	podBasedMutatingWebhooks   = []string{"mdeployment.kb.io", "mpod.kb.io", "mstatefulset.kb.io"}
+	frameworkWebhooks = map[string]string{
+		"pod":                           "pod",
+		"batch/job":                     "job",
+		"kubeflow.org/mpijob":           "mpijob",
+		"ray.io/rayjob":                 "rayjob",
+		"ray.io/raycluster":             "raycluster",
+		"jobset.x-k8s.io/jobset":        "jobset",
+		"kubeflow.org/mxjob":            "mxjob",
+		"kubeflow.org/paddlejob":        "paddlejob",
+		"kubeflow.org/pytorchjob":       "pytorchjob",
+		"kubeflow.org/tfjob":            "tfjob",
+		"kubeflow.org/xgboostjob":       "xgboostjob",
+		"deployment":                    "deployment",
+		"statefulset":                   "statefulset",
+		"kueue.x-k8s.io/clusterqueue":   "clusterqueue",
+		"kueue.x-k8s.io/workload":       "workload",
+		"kueue.x-k8s.io/resourceflavor": "resourceflavor",
+		"kueue.x-k8s.io/cohort":         "cohort",
```

Comment:
I believe @PannagaRao work on not installing the alpha CRDs would mean that we should not have Cohorts yet.

Oh, we probably still have the webhooks..

Reply:
>Oh, we probably still have the webhooks..

exactly.